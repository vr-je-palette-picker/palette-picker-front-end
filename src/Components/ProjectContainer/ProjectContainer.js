import React from 'react';
import { MdDeleteForever } from 'react-icons/md';
import PaletteCard from './../PaletteCard/PaletteCard.js';
import { deleteProject } from './../../utils/apiCalls/apiCalls.js'

const ProjectContainer = ({ project, fetchProjects }) => {
  const paletteCards = project.palettes.map((palette, index) => {
    return <PaletteCard key={index} palette={palette} />;
  });
  const handleDelete = async (id) => {
    await deleteProject(id)
    return await fetchProjects()
  }

  return (
    <section className='ProjectContainer'>
      <header className='ProjectContainer__header'>
        <h2 className='ProjectContainer__h2--project-name'>
          {project.project}
        </h2>
        <MdDeleteForever
          className='ProjectContainer__header--delete-icon'
          onClick={() => handleDelete(project.id)}
        />
      </header>
      <section className='ProjectContainer__section'>{paletteCards}</section>
    </section>
  );
};

export default ProjectContainer;
