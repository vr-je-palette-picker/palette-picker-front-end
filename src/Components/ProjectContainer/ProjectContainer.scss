.ProjectContainer {
  border: 2px lightgrey solid;
  padding: 27px;
  margin: 20px 20px 40px 20px;
  section {
    display: flex;
    height: 350px;
    overflow-y: scroll;
  }
}

.ProjectContainer__header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
    h2 {
      font-size: 40px;
      font-weight: 200;
      margin: 10px 0;
      padding-left: 20px;
    }
}

.ProjectContainer__header--delete-icon {
  cursor: pointer;
  height: 20px;
  padding: 10px;
  width: 20px;
}