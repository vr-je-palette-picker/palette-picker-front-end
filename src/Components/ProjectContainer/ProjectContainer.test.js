import React from 'react';
import { shallow } from 'enzyme';
import ProjectContainer from './ProjectContainer';

describe('ProjectContainer', () => {
  let wrapper;
  let project = {
    palettes: [{}, {}, {}]
  };
  let key = 1;

  beforeEach(() => {
    wrapper = shallow(<ProjectContainer key={key} project={project} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
