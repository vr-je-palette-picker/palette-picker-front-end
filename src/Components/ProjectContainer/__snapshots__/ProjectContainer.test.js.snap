// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer should match the snapshot 1`] = `
<section
  className="ProjectContainer"
>
  <header
    className="ProjectContainer__header"
  >
    <h2
      className="ProjectContainer__h2--project-name"
    />
    <MdDeleteForever
      className="ProjectContainer__header--delete-icon"
      onClick={[Function]}
    />
  </header>
  <section
    className="ProjectContainer__section"
  >
    <PaletteCard
      key="0"
      palette={Object {}}
    />
    <PaletteCard
      key="1"
      palette={Object {}}
    />
    <PaletteCard
      key="2"
      palette={Object {}}
    />
  </section>
</section>
`;
